### PR TITLE
Update configs for additional partitions and flash vendor_boot.img

### LINKED ISSUE #303: Add supprt for flashing vendor_boot.img
Some, especially Android 13 ROMs require flashing vendor_boot before moving on.

Example: Google Pixel 5 (redfin): https://wiki.lineageos.org/devices/redfin/install

Steps:
- Add a tooling function `fastboot flash vendor_boot {vendor_boot.img}` (and add it to the required code files)
- Add selection menues to the select views.

### PR DESCRIPTION
**Pull request by tsterbak**
The PR adds:
- updates for some OnePlus and Pixel configs with make use of the flashing of additional partitions if needed for Android 13
- Allow flashing of the vendor_boot partition
- Add --disable-verity --disable-verification  to flash vbmeta command
- support for alioth (Redmi K40 / Mi 11X / Poco F3) (thanks to @anon1892)

Resolves #303 and #304 .

### CODE REVIEW

File: openandroidinstaller/assets/configs/Mi439.yaml [hunk 1]
```diff
@@ -53,7 +53,7 @@ steps:
       content: >
         Install additional partitions selected before by pressing 'Confirm and run'. Once it's done continue.
         
-        Note : If you have not selected this partition, it will do nothing.
+        Note : If you have not selected additional partitions it will do nothing and is save to continue.
```
Comment (MagicLike) [inline openandroidinstaller/assets/configs/Mi439.yaml:56]: This sentence sounds weird... Maybe it's better like this?: "[...] and it is save [...]"

File: openandroidinstaller/assets/configs/avicii.yaml [hunk 2]
```diff
@@ -31,6 +34,12 @@ steps:
         adapting and repairing of the operating system.
         Make sure your device is turned on. You need to reboot into the bootloader again by pressing 'Confirm and run' here. Then continue.
       command: adb_reboot_bootloader
+    - type: call_button
+      content: >
+        Install additional partitions selected before by pressing 'Confirm and run'. Once it's done continue.
+        
+        Note : If you have not selected additional partitions it will do nothing and is save to continue.
```
Comment (MagicLike) [inline openandroidinstaller/assets/configs/avicii.yaml:41]: Same here and in all other configs :thinking:

File: openandroidinstaller/assets/configs/guacamoleb.yaml
Comment (MagicLike) [inline openandroidinstaller/assets/configs/guacamoleb.yaml:1]: Is the missing of the `fastboot_flash_additional_partitions` part on purpose or a mistake?

**Review by MagicLike — CHANGES_REQUESTED** [review]

File: openandroidinstaller/assets/configs/guacamoleb.yaml
Comment (tsterbak) [reply openandroidinstaller/assets/configs/guacamoleb.yaml:1]: Good catch! It's a mistake.

**Comment by tsterbak** [comment]
Actually, I wonder if this whole extra step for flashing additional partitions can be pulled into the "boot recovery" step all together. Maybe I'll try that

**Comment by tsterbak** [comment]
@anon1892 I tried to pull flash additional partitions in the fastboot_flash_recovery function, to simplify the steps and remove the necessity to perform an empty step, if you don't need the partitions now.

The change is currently only implemented for `Mi439` for testing purposes. Do you have the device available and could give this a test? Then I would implement it for more devices.

**Comment by tsterbak** [comment]
> @tsterbak Yes, I may test it next week, I'm busy now...

No worries :) looking forward to hear from you :)

**Comment by anon1892** [comment]
Hello!

I just tested this PR to install Lineage 20 from MiUI on my `pine`, and it worked fine!
If I select “recovery already flashed”, I can't choose additional partitions, but I think it's the wanted behavior (just to be sure)

**Comment by anon1892** [comment]
I was looking at [Lineage documentation](https://wiki.lineageos.org/devices/alioth/install/variant1#installing-a-custom-recovery-using-fastboot) for #275, and this line
https://github.com/openandroidinstaller-dev/openandroidinstaller/blob/95e08f7328e67347e1d423acba921a1ffad4bd2f/openandroidinstaller/tooling.py#L475
have to be replaced with `fastboot reboot-recovery`

I tested on my lavender, it to the same (in both cases I have to push Vol+ to boot, the line is equivalent to `fastboot reboot`). But we have to be sure that all phones support `fastboot reboot-recovery` at least the same way they support `fastboot reboot recovery`...

**Comment by tsterbak** [comment]
Thanks for figuring that out!

Do you want to go through the configs of the Xiaomi devices and update that? 
For the `fastboot reboot-recovery` I would advise to create a new function analogous to `fastboot reboot recovery` and use that if required.

**Comment by anon1892** [comment]
I looked at it on the internet, both look interchangeable and can be used in the same way. On my tested devices, both are equivalent. Since this function is only used by Xiaomi devices, the only regression it can create is for `vayu` : it's a “tested” device, but not by me...
I don't think it's necessary to create a second function for that.

> Do you want to go through the configs of the Xiaomi devices and update that?

What ?

**Comment by tsterbak** [comment]
> What ?

Sorry for not being precise, I didn't get it properly. If that method is only used for Xiaomi devices, we should put a comment there in the function and then I guess its fine to change it to `reboot-recovery`.

The other question I had, was if the `reboot recovery` is even needed if you have to press the buttons either way. I have a MI 9 Lite (pyxis) device now and tested it here: [openandroidinstaller/assets/configs/pyxis.yaml](https://github.com/openandroidinstaller-dev/openandroidinstaller/pull/326/files#diff-81d204d73cd727595da6dbb4949fb566faa974cd21533cd8bfee642071408cb0)
For this device reboot recovery and reboot-recovery do nothing after flashing recovery, so I have to press the buttons to get it going. Is this the same for your devices?

**Comment by anon1892** [comment]
Yes, for now it's only used by Xiaomi devices ([proof](https://github.com/search?q=repo%3Aopenandroidinstaller-dev%2Fopenandroidinstaller+fastboot_reboot_recovery+path%3A%2F%5Eopenandroidinstaller%5C%2Fassets%5C%2Fconfigs%5C%2F%2F&type=code))

Yes, for my `lavender` it's the same : I always need to press Vol+. For my `dandelion`, both works without pressing any button. For `alioth`, `reboot-recovery` seems to be working the right way according to Lineage documentation.

File: openandroidinstaller/assets/configs/Mi439.yaml (hunk 1 above)
Comment (tsterbak) [reply openandroidinstaller/assets/configs/Mi439.yaml]: This is completely removed now, since this step is no longer required.

File: README.md [hunk 3]
```diff
@@ -183,7 +183,7 @@ Xiaomi | Redmi 7A / 8 / 8A / 8A Dual | [Mi439](https://wiki.lineageos.org/device
 Xiaomi | Redmi Note 8 / 8T | [ginkgo](https://wiki.lineageos.org/devices/ginkgo) / willow |  | untested
 Xiaomi | Redmi Note 8 Pro | begonia |  | untested
 Xiaomi | Redmi 9A / 9C / 9AT / 9i / 9A Sport / 10A / 10A Sport | garden / dandelion / blossom / angelican | | tested
-Xiaomi | Mi 9T / Redmi K20 | [davinci](https://wiki.lineageos.org/devices/davinci) / davinciin |  | untested
+Xiaomi | Mi 9T / Redmi K20 | [davinci](https://wiki.lineageos.org/devices/davinci) / davincin |  | untested
```
Comment (anon1892) [inline README.md]: I'm a pro for doing typos, but this one wasn't!

Comment (tsterbak) [reply README.md]: I'm fixing it :)

**Pull request merged**